feat: add a sticky ad slot to the Advertising wizard

dkoo opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Make sure to test alongside https://github.com/Automattic/newspack-plugin/pull/812.

Closes #1199.

### How to test the changes in this Pull Request:

Follow testing instructions in https://github.com/Automattic/newspack-plugin/pull/812.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

laurelfulford reviewed (changes requested):
Overall this is looking good, @dkoo, and working as described!

In my testing, I spotted a couple inconsistencies between the AMP and non-AMP versions - nothing major though:

**When AMP is disabled:**

* The sticky ad location is scaled up to fit taller ads:

![image](https://user-images.githubusercontent.com/177561/104649483-a8e3fa80-5669-11eb-961f-9ac035d14352.png)

* The ads are not centered: 

![image](https://user-images.githubusercontent.com/177561/104651473-8f907d80-566c-11eb-815c-b0051f9ff617.png)

**When AMP is enabled:**

* The sticky space is not scaled up to fit taller ads (I don't thing we want to encourage very tall ads in this location, but I'm not sure what the intended behaviour should be):

![image](https://user-images.githubusercontent.com/177561/104649774-0f691880-566a-11eb-9aca-4591d23d44fe.png)

* There is a bit of space above the ad, that doesn't appear when AMP is disabled, and seems unbalanced, since it's only at the top. It's coming from a separate element, `amp-sticky-ad-top-padding`, so might be mandatory, I'm not sure. If yes, it might be nice to add it to the bottom, just to balance it out a bit. 

![image](https://user-images.githubusercontent.com/177561/104651626-c23a7600-566c-11eb-85d1-81052c1a6cc6.png)

Otherwise, things worked as expected, and displayed consistently across different browsers.

dkoo commented:
> The sticky space is not scaled up to fit taller ads (I don't thing we want to encourage very tall ads in this location, but I'm not sure what the intended behaviour should be):

I can answer this one! `amp-sticky-ad` has a max-height of 100px, per the [specs](https://amp.dev/documentation/components/amp-sticky-ad/). If we want to maintain parity across AMP vs. non-AMP, we can enforce this max height for non-AMP, too—should we?

laurelfulford commented:
> If we want to maintain parity across AMP vs. non-AMP, we can enforce this max height for non-AMP, too—should we?

Ah, that makes sense! 

My vote would be to add a max-height for the non-AMP version too -- that way they'll behave consistently, and we can use it as a way to discourage adding a huuuuge ad there, too :)

dkoo commented:
> The ads are not centered

Doh, the container needs `display: flex`, not `block`. Fixed in 644f93b.

> There is a bit of space above the ad, that doesn't appear when AMP is disabled, and seems unbalanced, since it's only at the top. It's coming from a separate element, amp-sticky-ad-top-padding, so might be mandatory, I'm not sure. If yes, it might be nice to add it to the bottom, just to balance it out a bit.

This is coming from the `amp-sticky-ad` element, but it looks like we can hide it using CSS. I've removed it so there's no padding either above or below the creative in both AMP and non-AMP. (Also 644f93b)

> My vote would be to add a max-height for the non-AMP version too -- that way they'll behave consistently, and we can use it as a way to discourage adding a huuuuge ad there, too :)

Done! 644f93b again.

laurelfulford commented on newspack-theme/sass/plugins/newspack-ads.scss:
```diff
@@ -14,6 +14,78 @@
+		}
+
+		> * {
+			margin: 0;
```
This is getting into super nitpicky territory since this space is not intended for tall ads, but the max-height on the non-AMP ads is making it possible to create an ad that can never be dismissed -- it's pushing the X out of view as well as keeping the ad contained:

![image](https://user-images.githubusercontent.com/177561/104665182-9bd50480-5685-11eb-8663-00ba001b3b36.png)

If you remove the `overflow-y` and `max-height` from `.sticky` and move the max-height here, it makes sure the too-tall add is contained without pushing the X out of sight: 

```suggestion
			margin: 0;
			max-height: 100px;
```

![image](https://user-images.githubusercontent.com/177561/104665238-bad39680-5685-11eb-8b0d-30dd219131e5.png)

dkoo replied:
Ah, good suggestion—this works as you describe. Updated in 87af381.

laurelfulford reviewed (changes requested):
Thanks @dkoo! Everything looks good -- just added one more nitpicky thing around the too-tall ads behaviour. If it's not as easy to fix as I assumed, I don't think it's a blocker -- just a weird thing that would happen if you did accidentally assign a tall ad to this spot :)

laurelfulford reviewed (approved):
Thanks Derrick! Looks good 🚢
